source/unified-test-format/unified-test-format.rst

```diff
@@ -2493,7 +2559,229 @@ An example of this operation follows::
+times so far in the test.
 
+The following arguments are supported:
+
+- ``client``: Required string. Client entity whose events the runner will be
+  counting.
+
+- ``event``: Required `expectedEvent`_ object. The event which the test runner
+  will be counting on the provided client. The assertions laid out in
+  `expectedEvent`_ MUST be used to determine if an observed event matches
+  ``event``. ``event`` SHOULD have an event type that was included in the
+  ``client``'s ``observeEvents`` field.
+
+- ``count``: Required integer. The exact number of matching events that
+  ``client`` should have seen so far.
+
+For example, the following instructs the test runner to assert that
+a single PoolClearedEvent was published::
+
+      - name: assertEventCount
+        object: testRunner
+        arguments:
+          client: *client0
+          event:
+            poolClearedEvent: {}
+          count: 1
+
+recordTopologyDescription
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The "recordTopologyDescription" operation instructs the test runner to retrieve
+the specified MongoClient's current `TopologyDescription <entity_topologydescription_>`_ and store it in
+the `Entity Map`_.
+
+The following arguments are suported:
+
+- ``client``: Required string. Client entity whose TopologyDescription will be recorded.
+
+- ``id``: Required string. The name with which the TopologyDescription will be
+  stored in the `Entity Map`_.
+
+For example::
+
+    - name: recordTopologyDescription
+      object: testRunner
+      arguments:
+        client: *client0
+        id: &postInsertTopology postInsertTopology
+
+assertTopologyType
+~~~~~~~~~~~~~~~~~~
+
+The ``assertTopologyType`` operation instructs the test runner to assert that
+the given `TopologyDescription <entity_topologydescription_>`_ has a particular TopologyType.
+
+The following arguments are supported:
+
+- ``topologyDescription``: Required string. TopologyDescription entity whose
+  TopologyType will be inspected.
+
+- ``topologyType``: Required string. Expected TopologyType for the
+  TopologyDescription. See `SDAM: TopologyType
+  <../server-discovery-and-monitoring/server-discovery-and-monitoring.rst#topologytype>`__
+  for a list of possible values.
+
+For example::
+
+  - name: assertTopologyType
+    object: testRunner
+    arguments:
+      topologyDescription: *postInsertTopology
+      topologyType: ReplicaSetWithPrimary
+
+waitForPrimaryChange
+~~~~~~~~~~~~~~~~~~~~
+
+The ``waitForPrimaryChange`` operation instructs the test runner to wait until
+the provided MongoClient discovers a different primary from the one in the
+provided `TopologyDescription <entity_topologydescription_>`_. If the provided
+TopologyDescription does not include a primary, then this operation will wait
+until the client discovers any primary.
+
+The following arguments are supported:
+
+- ``client``: Required string. See `commonOptions_client`_.
+
+  The clienty entity MUST the same one from which ``topologyDescription`` was
```

Comment:
```suggestion
  The client entity MUST the same one from which ``topologyDescription`` was
```

Typo and partial sentence? MUST the same what?

Reply:
I think this should read "MUST be", but @patrickfreed can confirm

Reply:
yep, should've been "MUST be". Fixed, and fixed the typo.